astropy/io/votable/connect.py

```diff
@@ -141,10 +143,12 @@ def write_table_votable(input, output, table_id=None, overwrite=False,
 
     # Tables with mixin columns are not supported
     if input.has_mixin_columns:
-        mixin_names = [name for name, col in input.columns.items()
-                       if not isinstance(col, input.ColumnClass)]
-        raise ValueError('cannot write table with mixin column(s) {0} to VOTable'
-                         .format(mixin_names))
+        # Only those columns which are instances of BaseColumn or Quantity can be written
+        unsupported_cols = input.columns.not_isinstance((BaseColumn, Quantity))
+        if unsupported_cols:
+            unsupported_names = [col.info.name for col in unsupported_cols]
+            raise ValueError('cannot write table with mixin column(s) {0} to VOTable'
+                         .format(unsupported_names))
```

Comment:
Indentation is not quite right, though if you remove the outer `if` statement, as suggested above, this will become OK.